### DISCUSSION TITLE
Improvements to the development workflow

### DISCUSSION THREAD

**Discussion by NicoWeio** [Arcticons-Team/Arcticons#1642] [Ideas]
This is an unstructured list of ideas I had. Make of it whatever you want; if you approve of some of them, I might do some work on them over the coming months.

- Add a way to get the `<package>/<activity>` specifier that is used in the AppFilter.xml. Currently, I send icon requests to myself, but after five times of doing that, I need to reset the app's data to reset the counter, which is suboptimal. (This might belong to _CandyBar_ upstream)
- Remove generated code, such as black/white icon versions from the repo. Generate them in GitHub Actions (as a build artifact) if necessary. The scripts would still work for local development. Maybe even build the app on Actions.
- Check/correct the formatting of AppFilter.xml in PRs, so that there are no more annoying whitespace changes.
- (This might be complicated:) Add a script that downloads every app and checks their manifest for launcher activities. This way, one could automatically find issues with renamed activities, where our icons are no longer applied.
- Move the app to a separate repo and use the icon repo as a submodue instead. I'm really not sure if it's worth it in this case, though.

**Comment by Donnnno** [comment]
Hey!
- Use [Icon Request](https://github.com/Kaiserdragon2/IconRequest/releases) or Icon Pusher instead of the build in way to gather appfilter entries! 
- Remove generated code, why and what? 
-  This might be a nice thing to have
- sounds complicated
- sounds complicated 2, also I don't see a need to do that, everything works fine now :) 

But feel free to take a look at it if you think it's better

**Comment by TotallyAvailable** [comment]
- have a fresh copy of the Apps in the Izzy and FDroid index around to keep the 'high FOSS coverage' in focus.
- ~Move "most requested" below "actively developed, yet still un-covered FOSS" (controversial&unnecessary, great for future users, bad for existing&waiting ones)~


Something I thought about doing myself and mentioned in a recent PR:

- split the piggybacked entries into their own, (only!) IF they do not create double listings of the same icon in a given (letter) category. 
Excluding Gallery, Contacts, Phone, ...for now. Some might absolutely deserve a separate listing/icon
If that messes with your flow more than it help keeping track of things, I'll obviously drop this one.

- A document listing new unused (information needed), existing unused, frequently broken and existing with region specific strings required.
Linking it among the other 'How to Contribute' information should allow to further widen the pool of possible contributions. Also more coverage before a new release. If it's there to be looked at right away instead of waiting for feedback after a new release.
